libsolidity/analysis/TypeChecker.cpp

```diff
@@ -913,6 +913,30 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 					var.location(),
 					"Cannot declare variable with void (empty tuple) type."
 				);
+			else if (valueComponentType->category() == Type::Category::RationalNumber)
+			{
+				string typeName = var.annotation().type->toString(true);
+				string extension;
+				if (auto type = dynamic_cast<IntegerType const*>(var.annotation().type.get()))
+				{
+					int numBits = type->numBits();
+					bool isSigned = type->isSigned();
+					if (isSigned)
+						numBits--;
+					extension = ", which can hold values up to " + string((u256(1) << numBits) - 1);
```

Comment:
It may be better displaying the range instead, that covers signed and unsigned types.